### PR TITLE
refactor: add cs_mapSocketToNode and cs_sendable_receivable_nodes to minimize cs_vNode contention (and document an undocumented lock requirement)

### PR DESCRIPTION
**Pull request by PastaPastaPasta**
## Issue being fixed or feature implemented
Reduce locking contention on cs_vNode

## What was done?
added new mutex

## How Has This Been Tested?
Building running locally on testnet

## Breaking Changes
None

## Checklist:
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added or updated relevant unit/integration/functional/e2e tests
- [ ] I have made corresponding changes to the documentation
- [x] I have assigned this pull request to a milestone

### CODE REVIEW

**Comment by DashCoreAutoGuix** [comment]
Guix Automation has began to build this PR tagged as v21.0.0-devpr5956.ef2162a1. A new comment will be made when the image is pushed.

**Comment by DashCoreAutoGuix** [comment]
Guix Automation has completed; a release should be present here: https://github.com/dashpay/dash-dev-branches/releases/tag/v21.0.0-devpr5956.ef2162a1. The image should be on dockerhub soon.

**Review by UdjinM6 — COMMENTED** [review]
one nit + would be nice to have slightly cleaner PR title and description ;)

**Review by UdjinM6 — DISMISSED** [review]
utACK

**Comment by PastaPastaPasta** [comment]
Did a trivial force-push; prior CI was happy: https://gitlab.com/dashpay/dash/-/pipelines/1226758401

File: src/net.h [hunk 1]
```diff
@@ -1542,9 +1543,9 @@ friend class CNode;
     int epollfd{-1};
 #endif
 
-    /** Protected by cs_vNodes */
-    std::unordered_map<NodeId, CNode*> mapReceivableNodes GUARDED_BY(cs_vNodes);
-    std::unordered_map<NodeId, CNode*> mapSendableNodes GUARDED_BY(cs_vNodes);
+    Mutex cs_sendable_receivable_nodes;
```
Comment (knst) [inline src/net.h:1546]: there is not really many changes in bitcoin related to this mutex, so, I guess there won't be too much conflicts in future due to that optimization.

But check this commit from future:
```
commit d51d2a3bb5b0011efa1b4f1e2c9512a16ce9b347
Author: Sebastian Falbesoner <sebastian.falbesoner@gmail.com>
Date:   Sat Aug 28 20:57:52 2021 +0200

    scripted-diff: rename node vector/mutex members in CConnman
    
    -BEGIN VERIFY SCRIPT-
    
    ren() { sed -i "s/$1/$2/g" $3 $4 $5; }
    
    ren cs_vAddedNodes         m_added_nodes_mutex     src/net.h src/net.cpp
    ren vAddedNodes            m_added_nodes           src/net.h src/net.cpp
    ren cs_vNodes              m_nodes_mutex           src/net.h src/net.cpp src/test/util/net.h
    ren vNodesDisconnectedCopy nodes_disconnected_copy src/net.cpp
    ren vNodesDisconnected     m_nodes_disconnected    src/net.h src/net.cpp
    ren vNodesCopy             nodes_copy              src/net.cpp
    ren vNodesSize             nodes_size              src/net.cpp
    ren vNodes                 m_nodes                 src/net.h src/net.cpp src/test/util/net.h
```
let's choose name accordingly new style with m_ prefix.

**Review by knst — COMMENTED** [review]
LGTM overall, but check a comment about naming variable

File: src/net.h (hunk 1 above)
Comment (PastaPastaPasta) [reply src/net.h:1546]: Hmm; feel free to propose a commit, but I don't think this is more than a nit

**Review by UdjinM6 — APPROVED** [review]
re-utACK

File: src/net.h (hunk 1 above)
Comment (knst) [reply src/net.h:1546]: will do when backport d51d2a3bb5b0011efa1b4f1e2c9512a16ce9b347

**Review by knst — APPROVED** [review]
LGTM

**Pull request merged**